### PR TITLE
Deployment Charts

### PR DESCRIPTION
**Pull request by Voxelot**
Adds helm configuration to deploy the block committer to k8s

### CODE REVIEW

File: deployment/charts/templates/deploy.yaml
Comment (liviudm) [inline deployment/charts/templates/deploy.yaml:10]: Ideally we want to use more labels under `selector` to make sure we target the right thing. When doing multiple deployments this is going to target all of them, unless `.Values.nameOverride` si specified and it's going to result in unexpected behaviours

File: deployment/charts/templates/deploy.yaml
Comment (liviudm) [inline deployment/charts/templates/deploy.yaml:5]: Please test this by doing 2 deployments in the same namespace. My feeling is that the second one is not going to work because it has the same object name

File: deployment/charts/templates/deploy.yaml
Comment (liviudm) [inline deployment/charts/templates/deploy.yaml:24]: Please test this by doing 2 deployments in the same namespace. My feeling is that the second one is not going to work because it has the same object name

File: deployment/charts/templates/deploy.yaml
Comment (liviudm) [inline deployment/charts/templates/deploy.yaml:26]: Not everyone wants to use NodePort. If you have nodes only in private networks, those will need to be exposed through a load balancer. Better give the option to the user to specify what the type of service

File: deployment/charts/templates/deploy.yaml
Comment (liviudm) [inline deployment/charts/templates/deploy.yaml:28]:
```
        app: {{ template "fuel-block-committer.name" . }}
        release: {{ .Release.Name }}
```

This is a better approach, otherwise it will forward traffic to all deployments in the namespace, which will create issues with network routing

File: deployment/charts/templates/deploy.yaml
Comment (liviudm) [inline deployment/charts/templates/deploy.yaml:38]: Please test this by doing 2 deployments in the same namespace. My feeling is that the second one is not going to work because it has the same object name

File: deployment/scripts/.env
Comment (liviudm) [inline deployment/scripts/.env:6]: IMO, this should be set as default in the Chart's default `values.yaml` file. Same for image tag, but never use `latest` for image tag

File: deployment/scripts/.env
Comment (liviudm) [inline deployment/scripts/.env:7]: never use `latest` for specifying image as per best practices

**Review by liviudm — CHANGES_REQUESTED** [review]

File: deployment/scripts/.env
Comment (Voxelot) [reply deployment/scripts/.env:7]: this is the base that's meant to be overwritten in by env overlays, so the values here are somewhat arbitrary. But I can set them to dummy values to ensure it doesn't accidentally succeed in an unexpected way.

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml:5]: I hardcoded the release name in the deployment script `service-deploy.sh`:
image

Due to this, subsequent workflows just update a singular deployment within the namespace. Should I try installing the charts under different release names or do you think this approach of hardcoding the release name via helm is sufficient?

**Comment by Voxelot** [comment]
@liviudm I updated all the places where there could be ambiguity in the labels to use the helm chart `Release.Name`. Since helm guarantees there can only be one release name per namespace, this ensures that in the event that two committers are deployed to the same namespace there won't be any confusion for things like the ServiceMonitor or NodePort.

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml:10]: added `release: {{ .Release.Name }}`

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml:42]: will leave this to a followup

Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml:42]: https://github.com/FuelLabs/fuel-block-committer/issues/42

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml:44]: done

File: deployment/scripts/.env
Comment (Voxelot) [reply deployment/scripts/.env:6]: We use envsubt on the default values of the chart in our current setup, so until we migrate to kustomize this is the way we provide defaults currently.

File: deployment/scripts/.env
Comment (Voxelot) [reply deployment/scripts/.env:7]: see https://github.com/FuelLabs/fuel-block-committer/pull/35/files#r1286509748

File: deployment/scripts/.env
Comment (Voxelot) [inline deployment/scripts/.env:10]:
```suggestion
fuel_committer_eth_chain_id="sepolia"
```

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml]: fixed by using the chart release name as an additional selector

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml]: changes to use the chart release name here to avoid conflicts

File: deployment/charts/templates/deploy.yaml
Comment (Voxelot) [reply deployment/charts/templates/deploy.yaml]: update to incorporate the chart release name

**Pull request merged**
